feat(react-scheduler): add horizontal grouping by resources (#2576)
DevExpress/devextreme-reactive: packages/dx-react-scheduler/docs/guides/grouping.md

# React Scheduler - Grouping

The React Scheduler allows you to group appointments by their [resources](../reference/resources.md). The group names are displayed by [GroupingPanel](../reference/grouping-panel.md) plugin.

> To change the group an appointment belongs to users can edit a corresponding appointment through the [AppointmentForm](../reference/appointment-form.md) or by drag-dropping the appointment with the help of [DragDropProvider](../reference/drag-drop-provider.md) plugin.

Reviewer: Seems like this paragraph is not required because it has been described in resources guide and in this guide below.

## Related Plugins

- [Resources](../reference/resources.md) - configures resources
- [GroupingState](../reference/grouping-state.md) - manages the grouping state
- [IntegratedGrouping](../reference/integrated-grouping.md) - performs built-in grouping

> - [GroupingPanel](../reference/grouping-panel.md) - renders the Grouping Panel

Reviewer: `GroupingPanel - renders the Grouping Panel` - good explanation 😂
Group Panel displays grouped resource titles

# React Scheduler - Grouping

Appointments can be grouped by [resources](../reference/resources.md). Group names are displayed on a grouping panel.

## Related Plugins

- [Resources](../reference/resources.md) - configures resources
- [GroupingState](../reference/grouping-state.md) - manages the grouping state
- [IntegratedGrouping](../reference/integrated-grouping.md) - implements grouping
- [GroupingPanel](../reference/grouping-panel.md) - displays resource titles (group names)

## Basic Usage

To group by resources, [configure them](./resources.md) first.

Grouping is implemented in the [IntegratedGrouping](../reference/integrated-grouping.md) plugin. Add this and the [GroupingState](../reference/grouping-state.md) plugin to enable this feature.

> `GroupingState` has a `grouping` property that accepts an array of resources to group by. Resource order in this array is important. For example, if the Room resource goes first, and Attendees goes second, appointments will be grouped in that order. If you leave the `grouping` property unspecified, the appointments are grouped by the [main resource](../reference/resources.md/#properties).

Reviewer 2: ```suggestion
`GroupingState` has a `grouping` property that accepts an array of resources to group by. The resource order in this array is important. For example, if the Room resource is first and Attendees second, appointments are grouped in that order. If the `grouping` property is unspecified, appointments are grouped by the [main resource](../reference/resources.md/#properties).
```

> Group names (resource titles) are displayed on a grouping panel. To show it, add the [GroupingPanel](../reference/grouping-panel.md) plugin.

Reviewer 2: ```suggestion
Group names (resource titles) are displayed in a grouping panel. To show it, add the [GroupingPanel](../reference/grouping-panel.md) plugin.
```

> A single appointment can appear in multiple groups if it is assigned to [multiple instances of a resource](./resources.md/#single-and-multiple-instance-resources). For example, an appointment assigned to three instances appears thrice - once in each group. The appointment's color is inherited from a particular resource instance. The `members` resource from the following demo illustrates this case:

Reviewer 2: ```suggestion
A single appointment can appear in multiple groups if it is assigned to [multiple instances of a resource](./resources.md/#single-and-multiple-instance-resources). For example, an appointment assigned to three instances appears once in each group. The appointment's color is inherited from a particular resource instance. The `members` resource from the following demo illustrates this case:
```

Reviewer 3: ```suggestion
A single appointment can appear in multiple groups if it is assigned to [multiple instances of a resource](./resources.md/#single-and-multiple-instance-resources). For example, an appointment assigned to three instances appears in three groups. The appointment's color is inherited from a particular resource instance. The `members` resource from the following demo illustrates this case:
```

Reviewer 2: Ok

.embedded-demo({ "path": "scheduler-grouping/basic", "showThemeSelector": true })

## Move Appointments Between Groups

> Users can drag-and-drop appointments between groups or use the appointment editing form to assign an an appointment to another resource (group). To enable these capabilities, configure [editing](./editing.md):

Reviewer 2: ```suggestion
Users can drag-and-drop appointments between groups or use the appointment edit form to assign an appointment to another resource (group). To enable this, configure [editing](./editing.md):
```
